internal/consumer/ackMgr.go

```diff
@@ -195,3 +204,25 @@ func (l *threadSafeSortedList) Add(value int64) (list.Address, error) {
 	}
 	return addr, err
 }
+
+// Reset blocks until the list is empty then sets lastValue to -1.
+func (l *threadSafeSortedList) Reset() {
+	doneC := make(chan struct{})
+	checkInterval := time.NewTicker(resetCheckInterval)
```

Comment:
stop the ticker before you exist this func to avoid a leak

Reply:
Done